packages/devtools-connect/src/connect.ts

```diff
@@ -504,16 +503,28 @@ async function connectMongoClientImpl({
               ...(clientOptions.proxy as DevtoolsProxyOptions),
               ...(ca ? { ca } : {}),
             },
-        clientOptions.applyProxyToOIDC ? undefined : 'mongodb://',
+        clientOptions.applyProxyToOIDC === true ? undefined : 'mongodb://',
       );
     cleanupOnClientClose.push(() => proxyAgent?.destroy());
 
-    if (clientOptions.applyProxyToOIDC) {
+    let oidcProxyOptions:
+      | AgentWithInitialize
+      | DevtoolsProxyOptions
+      | undefined;
+    if (clientOptions.applyProxyToOIDC === true) {
+      oidcProxyOptions = proxyAgent;
+    } else if (clientOptions.applyProxyToOIDC) {
+      oidcProxyOptions = clientOptions.applyProxyToOIDC;
+    }
+    if (!oidcProxyOptions && ca) {
+      oidcProxyOptions = { ca };
+    }
+    if (oidcProxyOptions) {
       clientOptions.oidc = {
+        customFetch: createFetch(
+          oidcProxyOptions,
+        ) as unknown as MongoDBOIDCPluginOptions['customFetch'],
         ...clientOptions.oidc,
-        ...addToOIDCPluginHttpOptions(clientOptions.oidc, {
-          agent: proxyAgent,
-        }),
```

Comment:
This is becoming necessary because openid-client@6.x only accepts a custom `fetch`, not custom HTTP options (which can more easily extended by adding more options to the object), so expanding `applyProxyToOIDC` to also accept a `DevtoolsProxyOptions` object allows us to keep doing pretty much the same thing when using this in Compass